sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -812,8 +814,17 @@ EventHubAsyncClient buildAsyncClient() {
 
         final TracerProvider tracerProvider = new TracerProvider(ServiceLoader.load(Tracer.class));
 
+        String identifier;
+        if (clientOptions != null && clientOptions instanceof AmqpClientOptions) {
+            String clientOptionIdentifier = ((AmqpClientOptions) clientOptions).getIdentifier();
+            identifier = clientOptionIdentifier == null ? UUID.randomUUID().toString() : clientOptionIdentifier;
+        } else {
+            identifier = UUID.randomUUID().toString();
+        }
```

Comment:
Are we not doing this in `EventProcessorClientBuilder`?

Reply:
We pass a `EventHubClientBuilder` to `EventProcessorClientBuilder`, user set `AmqpClientOptions` to `EventProcessorClientBuilder` will set to the `EventHubClientBuilder`, so processor client and the asyncClient in partition pumper will use the same identifier if user set a specific in `AmqpClientOptions`. Is it OK to do like this?

Reply:
@jsquire may know... Do we allow users to set a client identifier for EventProcessorClient? (ie. does that mean all the new underlying AMQP connections it creates, also use that identifier?)

Reply:
We do, and yes.    For the processor, this is especially important as we recommend users set a stable identifier for each processor instance.   Doing so allows the processor to enter recovery mode if the host crashes and reboots, since the partitions that it previously owned are associated with the identifier.   

Rather than going though load balancing and stealing random partitions because the old identifier still owns partitions until all of its ownership records expire, the rebooted processor can see that **it** should own those partitions, and the other processors in the group recognize this as well.   This prevents ownership from bouncing around, rewinds and duplication is reduced, and the rebooted instance can get back up to full speed quickly.

Reply:
The question on validation is cross-language. If all SDK does not do validation, Java probably can skip that as well. I just want to know what happens if I (or user) put something unreasonable. Does it crash the client? Or just make the connection unusable? I just want something predictable.

The later question on `AmqpClientOptions` is probably Java only. It might not be very important, as user can still call `client.getIdentifier` to get the value.

Reply:
Agree.

But I still expect developer of the PR aka @ZejiaJiang know:
1. what is valid, what is not valid, from client or service
2. what would Java client behave if invalid identifier is provided
(this is the simplest negative case in this PR)
with the knowledge we may decide whether to add to documentation.

Reply:
Yeah, I've asked Jesse before about identifier validation, From client side, we don't have any validation. We just let service to validate the identifier. But I think the best practice is I should confirm what's the validation in service side. We could validate the identifier before we send them to service to reduce the cost. I chose to align with .NET SDK before, because I think we don't want to let users get two different behaviors in two language SDK.

 Overall, I'll confirm what's the validation in service side at least, and decide should we add validation in client and add to documentation.

Reply:
I did a small test. if we set a 20KB string as identifier, the client will crash, but 2 KB works. It cause a BufferOverflowException when encode message to write into buffer.